job-simulation.rb

```diff
@@ -1,4 +1,59 @@
+# Since I was a little unclear on the directions (should I be writing a script with an output for a specific scenario? Or just a collection of methods that could potentially do these things?), Jeremy assigned me to make a class that can do it all.
+
+# If you want to try it out, open it up with irb -r ./job-simulation.rb and create an instance of the class to play with (hr = HiringProcess.new). Then run hr.first_hires to set up the first part of the process.
+
 require './Stack.rb'
 require './Queue.rb'
 
+class HiringProcess
+  attr_accessor :waitlist, :workers
+
+  def initialize
+    @waitlist = Queue.new
+    @workers = Stack.new
+  end
+
+# Must run this method on your new instance to do that initial hiring.
+# Didn't know how to use the initializer for this without also editing the initializers of the Stack class and the Queue class so that they could take parameters. (I thought those classes are supposed to remain untouched so their use is not specific to this problem.) So I just created this company_start method to handle those first six people. Should I have copied the contents of this method into the initializer? Does it matter?
+  def first_hires
+    (1..10).each do |number|
+      add_to_waitlist(number)
+    end
+    hire(6)
+    self
+  end
```

Comment:
This shouldn't need to be a separate call that we need to make right after object initialization. Instead call this method directly from the initialize method, like:

```
def initialize
  @waitlist = Queue.new
  @workers = Stack.new
  first_hires
end
```

But alternatively or additionally, it'd be awesome if you pass the number of workers and hires as parameters to this method. Maybe even do some flexible initialization: http://www.rubyist.net/~slagell/ruby/objinitialization.html